azure-toolkit-libs/azure-toolkit-containerregistry-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerregistry/AzureContainerRegistryModule.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.containerregistry;
+
+import com.azure.resourcemanager.containerregistry.ContainerRegistryManager;
+import com.azure.resourcemanager.containerregistry.models.Registries;
+import com.azure.resourcemanager.containerregistry.models.Registry;
+import com.microsoft.azure.toolkit.lib.common.model.AbstractAzResourceModule;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.telemetry.AzureTelemetry;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class AzureContainerRegistryModule extends AbstractAzResourceModule<ContainerRegistry, AzureContainerRegistryResourceManager, Registry> {
+
+    public static final String NAME = "registries";
+
+    public AzureContainerRegistryModule(@NotNull AzureContainerRegistryResourceManager parent) {
+        super(NAME, parent);
+    }
+
+    @Nonnull
+    @Override
+    protected Stream<Registry> loadResourcesFromAzure() {
+        return this.getClient().list().stream();
+    }
+
+    @Nullable
+    @Override
+    protected Registry loadResourceFromAzure(@Nonnull String name, String resourceGroup) {
+        return this.getClient().getByResourceGroup(resourceGroup, name);
+    }
+
+    @Override
+    protected void deleteResourceFromAzure(@Nonnull String resourceId) {
+        this.getClient().deleteById(resourceId);
+    }
+
+    @Override
+    protected ContainerRegistry newResource(@NotNull Registry registry) {
+        return new ContainerRegistry(registry, this);
+    }
+
+    @Override
+    @AzureOperation(name = "resource.draft_for_create.resource|type", params = {"name", "this.getResourceTypeName()"}, type = AzureOperation.Type.SERVICE)
+    protected ContainerRegistryDraft newDraftForCreate(@Nonnull String name, String resourceGroup) {
+        AzureTelemetry.getContext().setProperty("resourceType", this.getFullResourceType());
+        AzureTelemetry.getContext().setProperty("subscriptionId", this.getSubscriptionId());
```

Comment:
setting `resourceType` & `subscriptionId` is no longer required, they would be set automatically in the aspect of `@AzureOperation`.